oshdb-api/src/main/java/org/heigit/bigspatialdata/oshdb/api/mapreducer/backend/MapReducerIgniteLocalPeek.java

```diff
@@ -226,34 +241,73 @@ public boolean isActive() {
       return this.notCanceled;
     }
 
-    public abstract S execute(Ignite node);
+    private class CellKeysIterator implements Iterator<Long> {
+      Iterator<Pair<CellId, CellId>> cellIdRanges;
+      ArrayList<Long> buffer;
+
+      // a buffer of about 1M interleaved cellIds
+      static final int BUFFER_SIZE = 1_000_000;
```

Comment:
Better to you a buffer-size with a multiple of 1024, because it seems 1024 this is the buffer size for parallel streams.